packages/@sanity/migrate/src/runner/utils/__tests__/batchMutations.test.ts

```diff
@@ -0,0 +1,158 @@
+import {at, createIfNotExists, createOrReplace, patch, set} from '@bjoerge/mutiny'
+import {Mutation} from '@sanity/client'
+import {batchMutations} from '../batchMutations'
+
+function byteLength(obj: unknown) {
+  return JSON.stringify(obj).length
+}
+
+describe('mutation payload batching', () => {
+  test('when everything fits into a single mutation request', async () => {
+    const first = {createIfNotExists: {_id: 'foo', _type: 'something', bar: 'baz'}}
+    const second = {patch: {id: 'foo', set: {bar: 'baz'}}}
+    const gen = async function* () {
+      yield first
+      yield second
+    }
+
+    const firstSize = JSON.stringify(first).length
+    const secondSize = JSON.stringify(second).length
+
+    const it = batchMutations(gen(), firstSize + secondSize)
+
+    expect(await it.next()).toEqual({value: [first, second], done: false})
+    expect(await it.next()).toEqual({value: undefined, done: true})
+  })
+
+  test('when max batch is not big enough to fit all values', async () => {
+    const first = {createIfNotExists: {_id: 'foo', _type: 'something', bar: 'baz'}}
+    const second = {patch: {id: 'foo', set: {bar: 'baz'}}}
+    const gen = async function* () {
+      yield first
+      yield second
+    }
+
+    const firstSize = JSON.stringify(first).length
+
+    const it = batchMutations(gen(), firstSize)
+
+    expect(await it.next()).toEqual({value: [first], done: false})
+    expect(await it.next()).toEqual({value: [second], done: false})
+    expect(await it.next()).toEqual({value: undefined, done: true})
+  })
+  test('when each mutation is smaller then max batch size', async () => {
```

Comment:
Should this read "bigger", instead of "smaller"?

Reply:
Yes 🙈